# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/loader/ModuleLoaderBase.h
# Commit: cc6bc73fbbfd
# Full Hash: cc6bc73fbbfd2dd796ae51f2b85c7e9c24ae2686
# Author: Alexandru Marc <amarc@mozilla.com>
# Date: 2025-09-25 05:12 +0300
# Regressor Bug: 1968890
# File Overlap Count: 14
# Overlapping Files: dom/base/test/jsmodules/chrome.toml, dom/base/test/jsmodules/test_dynamicImportErrorMessage.html, dom/script/ScriptLoadContext.cpp, dom/script/ModuleLoader.cpp, dom/base/test/jsmodules/import_404.mjs
# Description:
#   Revert "Bug 1968890 - Part 9: Add test for dynamic import. r=jonco" for causing multiple wpt failures
# ==============================================================================

diff -r e1c5be04b3a0 -r cc6bc73fbbfd js/loader/ModuleLoaderBase.h
--- a/js/loader/ModuleLoaderBase.h	Wed Sep 24 23:17:54 2025 +0000
+++ b/js/loader/ModuleLoaderBase.h	Thu Sep 25 05:12:32 2025 +0300
@@ -549,10 +549,23 @@
                                              Handle<Value> aHostDefined,
                                              Handle<Value> aError);
   static bool OnLoadRequestedModulesResolved(ModuleLoadRequest* aRequest);
-  static bool OnLoadRequestedModulesRejected(JSContext* aCx,
-                                             ModuleLoadRequest* aRequest,
+  static bool OnLoadRequestedModulesRejected(ModuleLoadRequest* aRequest,
                                              Handle<Value> aError);
 
+  /**
+   * Shorthand Wrapper for JSAPI FinishDynamicImport function for the reject
+   * case where we do not have `aEvaluationPromise`. As there is no evaluation
+   * Promise, JS::FinishDynamicImport will always reject.
+   *
+   * @param aRequest
+   *        The module load request for the dynamic module.
+   * @param aResult
+   *        The result of running ModuleEvaluate -- If this is successful, then
+   *        we can await the associated EvaluationPromise.
+   */
+  void FinishDynamicImportAndReject(ModuleLoadRequest* aRequest,
+                                    nsresult aResult);
+
   void RemoveDynamicImport(ModuleLoadRequest* aRequest);
 
   nsresult CreateModuleScript(ModuleLoadRequest* aRequest);
@@ -560,7 +573,6 @@
 
   void OnFetchSucceeded(ModuleLoadRequest* aRequest);
   void OnFetchFailed(ModuleLoadRequest* aRequest);
-  void Cancel(ModuleLoadRequest* aRequest);
 
   // The slot stored in ImportMetaResolve function.
   enum class ImportMetaSlots : uint32_t { ModulePrivateSlot = 0, SlotCount };
